electrumsv/nodeapi.py

```diff
@@ -711,6 +713,148 @@ async def jsonrpc_createrawtransaction_async(request: web.Request, request_id: R
+    parameter_values = transform_parameters(request_id, [ "txid", "include_watchonly" ],
+        parameters)
+    # Compatibility: Raises RPC_TYPE_ERROR for non-string (node check allows nulls).
+    txid = parameter_values[0]
+    if parameter_values[0] is not None:
+        node_RPCTypeCheckArgument(request_id, txid, str)
+        node_ParseHexV(request_id, "txid", txid)
+    # INCOMPATIBILITY: Raises RPC_INVALID_PARAMETER to indicate current lack of support for the
+    # "include_watchonly" parameter - it should always be null.
+    if len(parameter_values) > 1 and parameter_values[1] is not None:
+        raise web.HTTPInternalServerError(headers={"Content-Type": "application/json"},
+            text=json.dumps(ResponseDict(id=request_id, result=None,
+            error=ErrorDict(code=RPCError.PARSE_ERROR,
+            message="JSON value is not a null as expected"))))
+
+    account_history_output_rows: list[AccountHistoryOutputRow] = \
+        account.get_wallet().data.read_history_for_outputs(account.get_id(),
+            transaction_hash=hex_str_to_hash(txid))
+
+    transaction_info_obj = dict[bytes, TransactionInfo]()
+
+    # Current transaction
+    previous_transaction_hash = ""
+    details: list[TransactionDetails] = []
+    for row in account_history_output_rows:
+        # Build the transaction details
+
+        # "immature", "orphan", "generate" not implemented yet
+        category = "receive" if row.value > 0 else "send"
+
+        # INCOMPATIBILITY: The 'account' field is always "" as we do not support this feature in
+        # any way
+        # INCOMPATIBILITY: The 'comment' field relates to data that cannot be modified. There are
+        # therefore no plans to support this property.
+        # INCOMPATIBILITY: The 'label' field relates to data that cannot be modified and the
+        # original bitcoind API does not add this property for vouts with no label/comment.
+        # There are therefore no plans to support this
+        # property.
+        # INCOMPATIBILITY: The 'abandoned' is always false as we always exclude deleted
+        # transactions from results  in the wallet proper.
+        # INCOMPATIBILITY: The involvesWatchonly field in details objects is never included as
+        # the node wallet API does not support watch-only accounts at this time.
+        transaction_details = TransactionDetails(
+            # address  # not implemented yet
+            # abandoned=None,  # not implemented yet
+            account="",
+            amount=row.value / 1e8,  # Convert from satoshis to bitcoins
+            category=category,
+            # fee=None,  # not implemented yet
+            vout=row.txo_index,
+        )
+
+        # Note: the SQL query has garaunteed ordering such that all the rows for a given transaction
+        # follow each other without gaps
+        if row.tx_hash == hex_str_to_hash(previous_transaction_hash):
```

Comment:
I cannot work out what is going on here. This whole loop seems to be constructed based on some reasoning that I do not understand.

It's this simple.
1. The user provides a txid.
2. All rows we get back will have that txid as the tx_hash.
3. We collect the details by iterating over them in the loop.
4. We construct the outer transaction object with common fields from either the first or last entry.
5. We fill in the aggregate fields based on all matched entries.
6. Done.